### DISCUSSION TITLE
Incorrect result using wgs84.latlon() ?

### DISCUSSION THREAD

**Discussion by brunobarwise** [skyfielders/python-skyfield#1013] [Q&A] [answered]
Hello !
I have started playing around with Skyfield, it's amazing.
I do find a strange result using wgs84.latlon() though, when I print the result : 

```
test_latlon = wgs84.latlon(-1.2345678, -2.3456789)
print(type(test_latlon))
print(test_latlon)

>>> <class 'skyfield.toposlib.GeographicPosition'>
>>> WGS84 latitude -1.2346 N longitude -2.3457 E elevation 0.0 m
```

Whatever ints or floats I put in, printing skyfield.toposlib.GeographicPosition objects always shows them as being N and E... Is this normal ? Is there a bug in my code ?

I'm a padawan coder, but the GeographicPosition parent class seems wrong ?

This code snippet from the Topos class :
```
class Topos(GeographicPosition):
    """Deprecated: use ``wgs84.latlon()`` or ``iers2010.latlon()`` instead."""

    def __init__(self, latitude=None, longitude=None, latitude_degrees=None,
                 longitude_degrees=None, elevation_m=0.0, x=0.0, y=0.0):

        if latitude_degrees is not None:
            pass
        elif isinstance(latitude, Angle):
            latitude_degrees = latitude.degrees
        elif isinstance(latitude, (str, float, tuple)):
            latitude_degrees = _ltude(latitude, 'latitude', 'N', 'S')
        else:
            raise TypeError('please provide either latitude_degrees=<float>'
                            ' or latitude=<skyfield.units.Angle object>'
                            ' with north being positive')
```

Tthis code snippet from the parent GeographicPosition  class :
```
@reify  # not @property, so users have the option of overwriting it
    def target_name(self):
        return '{0} latitude {1} N longitude {2} E elevation {3} m'.format(
            self.model.name,
            array2string(self.latitude.degrees, **_lat_options),
            array2string(self.longitude.degrees, **_lon_options),
            array2string(self.elevation.m, **_elev_options))
```

**Comment by brandon-rhodes** [comment, marked as answer]
> Whatever ints or floats I put in, printing skyfield.toposlib.GeographicPosition objects always shows them as being N and E... Is this normal ? Is there a bug in my code ?

Happily, that is completely normal. One can express, for example, the longitude 5°W also as the value –5°E, because the negative sign turns east into west. Internally, Skyfield uses negative to mean "west" and "south", instead of having a separate flag for east/west and north/south. So printing the raw values always will show north latitude and east longitude.

**Comment by brunobarwise** [comment]
🙇🏼‍♂️
